src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -316,6 +321,12 @@ public static Properties generateProxyPropertiesForJDBC() {
         proxyProperties.put(SFSessionProperty.PROXY_USER.getPropertyKey(), proxyUser);
         proxyProperties.put(SFSessionProperty.PROXY_PASSWORD.getPropertyKey(), proxyPassword);
       }
+
+      // Check if http.nonProxyHosts was set
+      final String nonProxyHosts = System.getProperty(NON_PROXY_HOSTS);
+      if (!isNullOrEmpty(nonProxyHosts)) {
+        proxyProperties.put(SFSessionProperty.NON_PROXY_HOSTS.getPropertyKey(), nonProxyHosts);
+      }
```

Comment:
Should we still need to check and put everything in `proxyProperties` if NON_PROXY_HOSTS = true?

Reply:
I'm assuming you meant if `USE_PROXY = true`, correct? 
Yes, someone may need to connect to host A and can only reach it by going through a proxy server, and at the same time, they need to connect to host B but can only reach that by going around the proxy server. `NON_PROXY_HOSTS `will contain the list of hosts that someone wants to connect to directly, so that has to be taken into account if it is set.